### PR TITLE
Show created at and data from in reports

### PR DESCRIPTION
**Pull request by noamblitz**
### Changes
Make it clearer that the data in the report is from a certain datetime and the report is created on another datetime

image

### Issue link
Closes https://github.com/minvws/nl-kat-coordination/issues/2356
Closes https://github.com/minvws/nl-kat-coordination/issues/2357

---

### Code Checklist
- [ ] All the commits in this PR are properly PGP-signed and verified.
- [ ] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have made corresponding changes to the documentation, if necessary.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/reports/views/base.py [hunk 1]
```diff
@@ -119,6 +120,7 @@ def get_report_types(self) -> List[ReportType]:
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context["observed_at"] = self.valid_time
+        context["created_at"] = datetime.now()
```
Comment (ammar92) [inline rocky/reports/views/base.py:123]: You might want to either convert this to a UTC timestamp, or whenever rendering this string also mention the timezone. Maybe this is the intended change, but it's not clear from the submitted screenshot

File: rocky/reports/templates/partials/report_header.html [hunk 2]
```diff
@@ -12,7 +12,10 @@
              alt="{% translate "OpenKAT" %}" />
         <h1>{% translate "Report for organization" %} {{ organization.name }}</h1>
         <p class="emphasized">
-            <strong>{% translate "Observed at:" %}</strong> {{ observed_at }} {{ TIME_ZONE }}
+            <strong>{% translate "Created with data from:" %}</strong> {{ observed_at }} {{ TIME_ZONE }}
+        </p>
+        <p>
+            <strong>{% translate "Created on:" %}</strong> {{ created_at }}
```
Comment (ammar92) [inline rocky/reports/templates/partials/report_header.html:18]:
```suggestion
            <strong>{% translate "Created on" %}</strong>: {{ created_at }}
```

File: rocky/reports/templates/partials/report_header.html [hunk 3]
```diff
@@ -12,7 +12,10 @@
              alt="{% translate "OpenKAT" %}" />
         <h1>{% translate "Report for organization" %} {{ organization.name }}</h1>
         <p class="emphasized">
-            <strong>{% translate "Observed at:" %}</strong> {{ observed_at }} {{ TIME_ZONE }}
+            <strong>{% translate "Created with data from:" %}</strong> {{ observed_at }} {{ TIME_ZONE }}
```
Comment (ammar92) [inline rocky/reports/templates/partials/report_header.html:15]:
```suggestion
            <strong>{% translate "Created with data from" %}</strong>: {{ observed_at }} {{ TIME_ZONE }}
```

File: rocky/reports/report_types/multi_organization_report/introduction.html [hunk 4]
```diff
@@ -15,7 +15,10 @@ <h1>Sector Report</h1>
             <p>{% translate "This is the OpenKAT Sector rapport." %}</p>
             <div>
                 <p>
-                    <strong>{% translate "Observed at:" %}</strong> {{ observed_at }} {{ TIME_ZONE }}
+                    <strong>{% translate "Created with date from:" %}</strong> {{ observed_at }} {{ TIME_ZONE }}
+                </p>
+                <p>
+                    {% translate "Created on:" %} <strong>{{ created_at }}</strong>
```
Comment (ammar92) [inline rocky/reports/report_types/multi_organization_report/introduction.html:21]:
```suggestion
                    {% translate "Created on" %}<strong>: {{ created_at }}</strong>
```

File: rocky/reports/report_types/multi_organization_report/introduction.html [hunk 5]
```diff
@@ -15,7 +15,10 @@ <h1>Sector Report</h1>
             <p>{% translate "This is the OpenKAT Sector rapport." %}</p>
             <div>
                 <p>
-                    <strong>{% translate "Observed at:" %}</strong> {{ observed_at }} {{ TIME_ZONE }}
+                    <strong>{% translate "Created with date from:" %}</strong> {{ observed_at }} {{ TIME_ZONE }}
```
Comment (ammar92) [inline rocky/reports/report_types/multi_organization_report/introduction.html:18]:
```suggestion
                    <strong>{% translate "Created with date from" %}</strong>: {{ observed_at }} {{ TIME_ZONE }}
```

File: rocky/reports/report_types/aggregate_organisation_report/introduction.html [hunk 6]
```diff
@@ -25,7 +25,10 @@ <h1>{% translate "OpenKAT Aggregate Report" %}</h1>
         <div>
             <p>{% translate "This is the OpenKAT report for organization" %} {{ organization.name }}</p>
             <p>
-                {% translate "Observed at:" %} <strong>{{ observed_at }} {{ TIME_ZONE }}</strong>
+                {% translate "Created with data from:" %} <strong>{{ observed_at }} {{ TIME_ZONE }}</strong>
+            </p>
+            <p>
+                {% translate "Created on:" %} <strong>{{ created_at }}</strong>
```
Comment (ammar92) [inline rocky/reports/report_types/aggregate_organisation_report/introduction.html:31]:
```suggestion
                {% translate "Created on" %}<strong>: {{ created_at }}</strong>
```

File: rocky/reports/report_types/aggregate_organisation_report/introduction.html [hunk 7]
```diff
@@ -25,7 +25,10 @@ <h1>{% translate "OpenKAT Aggregate Report" %}</h1>
         <div>
             <p>{% translate "This is the OpenKAT report for organization" %} {{ organization.name }}</p>
             <p>
-                {% translate "Observed at:" %} <strong>{{ observed_at }} {{ TIME_ZONE }}</strong>
+                {% translate "Created with data from:" %} <strong>{{ observed_at }} {{ TIME_ZONE }}</strong>
```
Comment (ammar92) [inline rocky/reports/report_types/aggregate_organisation_report/introduction.html:28]:
```suggestion
                {% translate "Created with data from" %}: <strong>{{ observed_at }} {{ TIME_ZONE }}</strong>
```

**Review by ammar92 — COMMENTED** [review]
Looks good in general. Colons should not be translated

**Comment by stephanie0x00** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.


### What works:
- Manually adding the 'created_at' to the URL as a parameter and playing with the values doesn't seem to impact the timestamping on the report. 

### What doesn't work:
- It is possible to create a report based on future data. The parameter `observed_at=2024-01-30` will result in:

future-created-date


- When changing the 'observed_at' parameter from `observed_at=2024-01-23` to `observed_at=2014-01-23` I get the following stack trace: 

date-reports


```
Environment:


Request Method: GET
Request URL: http://127.0.0.1:8000/en/ee/reports/generate-report/view/?observed_at=2014-01-23&plugin=nmap&plugin=dns-records&plugin=dns-sec&ooi=Hostname%7Cinternet%7Cmispo.es&report_type=name-server-report&report_type=dns-report

Django Version: 4.2.7
Python Version: 3.11.7
Installed Applications:
['whitenoise.runserver_nostatic',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.forms',
 'django_otp',
 'django_otp.plugins.otp_static',
 'django_otp.plugins.otp_totp',
 'two_factor',
 'account',
 'tools',
 'fmea',
 'crisis_room',
 'onboarding',
 'katalogus',
 'django_password_validators',
 'django_password_validators.password_history',
 'rest_framework',
 'tagulous',
 'compressor',
 'reports',
 'knox',
 'csp']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'whitenoise.middleware.WhiteNoiseMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.locale.LocaleMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django_otp.middleware.OTPMiddleware',
 'rocky.middleware.auth_required.AuthRequiredMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'rocky.middleware.onboarding.OnboardingMiddleware',
 'csp.middleware.CSPMiddleware']



Traceback (most recent call last):
  File "/app/rocky/octopoes/connector/octopoes.py", line 44, in _verify_response
    response.raise_for_status()
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

During handling of the above exception (404 Client Error: Not Found for url: http://octopoes_api:80/ee/tree?reference=Hostname%7Cinternet%7Cmispo.es&types=DNSRecord&types=Finding&depth=3&valid_time=2014-01-23+23%3A59%3A59.999999%2B00%3A00), another exception occurred:
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/rocky/reports/views/generate_report.py", line 145, in get
    return super().get(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 226, in get
    context = self.get_context_data(**kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/rocky/reports/views/generate_report.py", line 164, in get_context_data
    context["report_data"] = self.generate_reports_for_oois()
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/rocky/reports/views/generate_report.py", line 154, in generate_reports_for_oois
    data = report.generate_data(ooi, valid_time=self.valid_time)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/rocky/reports/report_types/dns_report/report.py", line 26, in generate_data
    tree = self.octopoes_api_connector.get_tree(
           
  File "/app/rocky/octopoes/connector/octopoes.py", line 152, in get_tree
    res = self.session.get(
          
  File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 602, in get
    return self.request("GET", url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/rocky/octopoes/connector/octopoes.py", line 64, in request
    self._verify_response(response)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/rocky/octopoes/connector/octopoes.py", line 48, in _verify_response
    raise ObjectNotFoundException(data["value"])
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Exception Type: ObjectNotFoundException at /en/ee/reports/generate-report/view/
Exception Value: Hostname|internet|mispo.es
```

**Comment by noamblitz** [comment]
Those things should be fixed in this pr: https://github.com/minvws/nl-kat-coordination/pull/2274

**Review by underdarknl — APPROVED** [review]

**Pull request merged**